src/hooks/useConnectRelatedVenues.ts

```diff
@@ -170,38 +171,43 @@ export const useConnectRelatedVenues: ReactHook<
 
   const makeEventsQueryConfig = (
     doc: string,
-    storeAs: string
-  ): SparkleRFQConfig =>
-    ({
-      collection: "venues",
-      doc,
-      subcollections: [{ collection: "events" }],
-      orderBy: ["start_utc_seconds", "asc"],
-      storeAs,
-    } as SparkleRFQConfig); // @debt a little hacky, but we're consciously subverting our helper protections;
+    storeAs: ValidFirestoreKeys
+  ): SparkleUseFirestoreConfigObject => ({
+    collection: "venues",
+    doc,
+    subcollections: [{ collection: "events" }],
+    orderBy: ["start_utc_seconds", "asc"],
+    storeAs,
+  });
 
   // Parent Events
-  const parentVenueEventsQuery: SparkleRFQConfig | undefined =
+  const parentVenueEventsQuery =
     parentId && withEvents
       ? makeEventsQueryConfig(parentId, "parentVenueEvents")
       : undefined;
 
   // Sibling Events
-  const siblingVenueEventsQueries: SparkleRFQConfig[] = withEvents
+  const siblingVenueEventsQueries = withEvents
     ? siblingVenues.map((sibling) =>
-        makeEventsQueryConfig(sibling.id, `siblingVenueEvents-${sibling.id}`)
+        makeEventsQueryConfig(
+          sibling.id,
+          `siblingVenueEvents-${sibling.id}` as ValidFirestoreKeys // @debt a little hacky, but we're consciously subverting our helper protections;
+        )
       )
     : [];
 
   // Sub Events
-  const subvenueEventsQueries: SparkleRFQConfig[] = withEvents
+  const subvenueEventsQueries = withEvents
     ? subvenues.map((subvenue) =>
-        makeEventsQueryConfig(subvenue.id, `subvenueEvents-${subvenue.id}`)
+        makeEventsQueryConfig(
+          subvenue.id,
+          `subvenueEvents-${subvenue.id}` as ValidFirestoreKeys // @debt a little hacky, but we're consciously subverting our helper protections;
+        )
       )
     : [];
 
   // Combine / filter for valid queries
-  const allValidQueries: SparkleRFQConfig[] = [
+  const allValidQueries: SparkleRFQConfig = [
```

Comment:
This feels more confusing than how it used to be.. probably would have named your 'fusion' type as `AnySparkleRFQConfig` or something specific about the props the function accepts.

Alternatively, given the change has already been done, this would probably be more correct/easy to understand as `SparkleRFQConfigObject[]`. 'Hiding' the array'ness of it makes it confusing, and I need to go any look up the original 'fusion type' to understand whats going on.